mne/_fiff/meas_info.py

```diff
@@ -833,6 +833,10 @@ def set_meas_date(self, meas_date):
         """
         from ..annotations import _handle_meas_date
 
+        _validate_type(
+            meas_date, (datetime.datetime, float, tuple, type(None)), "meas_date"
```

Comment:
If you do this magic
```suggestion
            meas_date, (datetime.datetime, "numeric", tuple, None), "meas_date"
```
you shouldn't need to change the `1`s below to `1.`'s or do `float(...)` etc.

We also have a nice translation from `None` to `type(None)` to avoid having to type that all over the place

Reply:
Okay, have added this and reverted those instances where I was changing ints to floats in the unit tests.